### DISCUSSION TITLE
What is your preferred ML development environment?

### DISCUSSION THREAD

**Discussion by kristopher-wood** [online-ml/river#698] [Show and tell]
Hi all!

I put this in Show and Tell not because I have anything to show, but because I hope you might!

What is your preferred ML Development environment? Which OS, language(s), IDE, libraries, etc make your life easier as an ML developer/researcher?

Currently I'm running Python 3.9 on Windows 10 using Spyder IDE in an Anaconda environment. I'm finding it very difficult to get some things working because I'm constantly fighting with Windows to get the various components to place nice together.

I hope some of you might have found a better way so that I can imitate your configurations!

**Comment by MaxHalford** [comment]
What exactly are you struggling with? A long time ago I used to develop with a Windows machine, but it wasn't fun. Now I'm on MacOS and it's great. The instructions in the `CONTRIBUTING.md` file should get do the trick. But I've never tried them on Windows.

**Reply by kristopher-wood** [reply]
Specifically the part that got me to throw in the towel on my current Dev environment is that I couldn't get windows/anaconda to see the path to the Chantilly run script after pip install. This isn't the first time I've run into path issues trying to get things working in Windows so I'm looking into what other people are using.

**Reply by kristopher-wood** [reply]
I got `chantilly run` to work in Windows by running it inside of an Ubuntu for Windows terminal! Now to see if I can get the rest of my dev tools to work in that environment....

**Reply by kristopher-wood** [reply]
Any tips on getting chantilly working with river instead of creme?

**Reply by MaxHalford** [reply]
Yes you'd have to change the Chantilly codebase to use learn_one instead of fit_one. But we're not maintaining because we're working on something new with @raphaelsty which will be more delightful and powerful :)

**Reply by kristopher-wood** [reply]
Do you need more developers for this more delightful something? 😉

I'm new to python but have over two decades professional development experience in general and learn new languages quickly. I’d be happy to contribute!

**Reply by MaxHalford** [reply]
For sure you'd be welcome to contribute! We first have to work on some core stuff, but once we release an alpha version there'll be loads of stuff to do.
